src/pages/ApproveAction/GenericApprovalScreen.tsx

```diff
@@ -69,34 +81,179 @@ export function GenericApprovalScreen() {
     action,
     network,
   });
+  const {
+    isGaslessOn,
+    setIsGaslessEligible,
+    getGaslessEligibility,
+    isGaslessEligible,
+    gaslessFundTx,
+    setIsGaslessOn,
+    isFundProcessReady,
+    fundTxHex,
+    fundTxDoNotRertyError,
+    setGaslessDefaultValues,
+    createGaslessOffscreen,
+    closeGaslessOffscreen,
+    solutionHex,
+    isGaslessFundStarted,
+    setIsGaslessFundStarted,
```

Comment:
Seems like there are multiple `is**` state variables which intend to store the current state of the funding process. 
However many of these cannot be true at the same time or being set to true implies that the other values are already true. 
Suggestion: use an enum instead to store the current status of the gasless flow. E.g:
Gasless: READY, ELIGIBLE, FUNDING_IN_PROGRESS, FUNDED, ERROR